src/app/core/services/machine-deployment.ts

```diff
@@ -82,4 +91,46 @@ export class MachineDeploymentService {
     const url = `${this._restRoot}/projects/${projectID}/clusters/${cluster}/machinedeployments/${mdId}/nodes/events`;
     return this._httpClient.get<Event[]>(url);
   }
+
+  showExternalMachineDeploymentDeleteDialog(
+    projectID: string,
+    cluster: ExternalCluster,
+    md: ExternalMachineDeployment
+  ): Observable<boolean> {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Delete External Machine Deployment',
```

Comment:
I think we can call this `Delete Machine Deployment` instead of `Delete External Machine Deployment` since we don't use `External Machine Deployment` in the UI (and message of this dialog).